sbom/formatted_reader_test.go

```diff
@@ -94,24 +96,77 @@ func testFormattedReader(t *testing.T, context spec.G, it spec.S) {
 		Expect(rerunBuffer).To(Equal(buffer))
 	})
 
-	it("writes the SBOM in the default SPDX format", func() {
-		buffer := bytes.NewBuffer(nil)
-		_, err := io.Copy(buffer, sbom.NewFormattedReader(bom, sbom.SPDXFormat))
-		Expect(err).NotTo(HaveOccurred())
+	context("writes the SBOM in SPDX format, with fields replaced for reproducibility", func() {
+		it("produces an SBOM", func() {
+			buffer := bytes.NewBuffer(nil)
+			_, err := io.Copy(buffer, sbom.NewFormattedReader(bom, sbom.SPDXFormat))
+			Expect(err).NotTo(HaveOccurred())
 
-		var spdxOutput spdxOutput
+			var spdxOutput spdxOutput
 
-		err = json.Unmarshal(buffer.Bytes(), &spdxOutput)
-		Expect(err).NotTo(HaveOccurred(), buffer.String())
+			err = json.Unmarshal(buffer.Bytes(), &spdxOutput)
+			Expect(err).NotTo(HaveOccurred(), buffer.String())
+
+			Expect(spdxOutput.SPDXVersion).To(Equal("SPDX-2.2"), buffer.String())
+
+			Expect(spdxOutput.Packages[0].Name).To(Equal("collapse-white-space"), buffer.String())
+			Expect(spdxOutput.Packages[1].Name).To(Equal("end-of-stream"), buffer.String())
+			Expect(spdxOutput.Packages[2].Name).To(Equal("insert-css"), buffer.String())
+			Expect(spdxOutput.Packages[3].Name).To(Equal("once"), buffer.String())
+			Expect(spdxOutput.Packages[4].Name).To(Equal("pump"), buffer.String())
+			Expect(spdxOutput.Packages[5].Name).To(Equal("wrappy"), buffer.String())
+
+			// Ensure documentNamespace and creationInfo.created have reproducible values
+			Expect(spdxOutput.DocumentNamespace).To(Equal("https://paketo.io/packit/dir/testdata-37a75df4-7cb2-5384-820a-d9ac961acccb"), buffer.String())
+			Expect(spdxOutput.CreationInfo.Created).To(BeZero(), buffer.String())
+
+			rerunBuffer := bytes.NewBuffer(nil)
+			_, err = io.Copy(rerunBuffer, sbom.NewFormattedReader(bom, sbom.SPDXFormat))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(rerunBuffer).To(Equal(buffer))
+		})
+
+		context("when SOURCE_DATE_EPOCH is set", func() {
+			var original string
+
+			it.Before(func() {
+				original = os.Getenv("SOURCE_DATE_EPOCH")
+				Expect(os.Setenv("SOURCE_DATE_EPOCH", "1659551872")).To(Succeed())
+			})
+
+			it.After(func() {
+				Expect(os.Setenv("SOURCE_DATE_EPOCH", original)).To(Succeed())
+			})
+
+			it("produces an SBOM with the given timestamp", func() {
+				buffer := bytes.NewBuffer(nil)
+				_, err := io.Copy(buffer, sbom.NewFormattedReader(bom, sbom.SPDXFormat))
+				Expect(err).NotTo(HaveOccurred())
+
+				var spdxOutput spdxOutput
```

Comment:
The unit tests for the FormattedReader need to be run sequentially because we're setting the env var, which is global state.

Reply:
Tests that set env vars have been made sequential